lib/beacon/authorization/behavior.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Beacon.Authorization.Behaviour do
+  @moduledoc """
+    Provides hooks into Beacon Authorization
+
+    ## Examples
+
+  """
+
+  # returns requestor_context
+  @callback get_requestor_context(payload :: any()) :: any()
+
+  # site, requestor_context, operation_context (needs type)
+  @callback authorized?(Beacon.Type.Site.t(), requestor_context :: any(), operation_context :: map()) :: boolean()
+  # requestor_context, operation_context (needs type)
+  @callback authorized?(requestor_context :: any(), operation_context :: map()) :: boolean()
```

Comment:
I don't love having two functions that differ just by arity. Can we either (a) combine these into `authorized?/3` and have a "reserved" atom for the first argument, or (b) rename `authorized?/2`?